tasklist/client/src/Tasks/CollapsiblePanel/index.tsx

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+
+import {useState} from 'react';
+import styles from './styles.module.scss';
+import sharedStyles from 'modules/styles/panelHeader.module.scss';
+import {Button} from '@carbon/react';
+import {SidePanelOpen, SidePanelClose, Filter} from '@carbon/react/icons';
+import cn from 'classnames';
+import {useSearchParams} from 'react-router-dom';
+import {type TaskFilters} from 'modules/hooks/useTaskFilters';
+import {SearchParamNavLink} from './SearchParamNavLink';
+import {prepareCustomFiltersParams} from 'modules/custom-filters/prepareCustomFiltersParams';
+import {getStateLocally} from 'modules/utils/localStorage';
+import difference from 'lodash/difference';
+import {useCurrentUser} from 'modules/queries/useCurrentUser';
+
+function getNavLinkSearchParam(options: {
+  currentParams: URLSearchParams;
+  filter: TaskFilters['filter'];
+  userId: string;
+}): string {
+  const CUSTOM_FILTERS_PARAMS = [
+    'state',
+    'followUpDateFrom',
+    'followUpDateTo',
+    'dueDateFrom',
+    'dueDateTo',
+    'assigned',
+    'assignee',
+    'taskDefinitionId',
+    'candidateGroup',
+    'candidateUser',
+    'processDefinitionKey',
+    'processInstanceKey',
+    'tenantIds',
+    'taskVariables',
+  ] as const;
+  const {filter, userId, currentParams} = options;
+  const {sortBy, ...convertedParams} = Object.fromEntries(
+    currentParams.entries(),
+  );
+  const values: Record<string, string> =
+    (sortBy === 'completion' && !['custom', 'completed'].includes(sortBy)) ||
+    sortBy === undefined
+      ? {filter}
+      : {filter, sortBy};
+  if (filter === 'custom') {
```

Comment:
I would extract this as a function which returns an object with the values to update. That way you don't have to repeat the params to delete.

```js
const {filter, userId, currentParams} = options;
const {sortBy, ...convertedParams} = Object.fromEntries( currentParams.entries() );
const customFilterParams = filter === 'custom' ? getCustomFilterParams( userId ) : {};

const updatedParams = new URLSearchParams({
  filter,
  ...convertedParams,
  ...values,
  ...customFilterParams,
});

if (sortBy !== undefined && sortBy !== 'completion') {
  updatedParams.set('sortBy', sortBy);
}
if (filter === 'completed') {
  updatedParams.set('sortBy', 'completion');
}

difference(
  CUSTOM_FILTERS_PARAMS,
  Object.keys(customFiltersParams),
).forEach((param) => {
  updatedParams.delete(param);
});

return updatedParams.toString();
```